### PR TITLE
[CI] Use `--all-targets` with `cargo check`

### LINKED ISSUE #95: [CI] Use cargo check --all-targets flag

### PR DESCRIPTION
**Pull request by jswrenn**
Ensures that examples, benchmarks, etc. are also checked.

Fixes #95.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -122,7 +122,7 @@ jobs:
         key: "${{ env.ZC_TOOLCHAIN }}-${{ matrix.target }}-${{ env.FEATURES_HASH }}-${{ hashFiles('**/Cargo.lock') }}"
 
     - name: Check
-      run: cargo +${{ env.ZC_TOOLCHAIN }} check --manifest-path ${{ matrix.manifest-path }} --target ${{ matrix.target }} --features "${{ matrix.features }}" --verbose
+      run: cargo +${{ env.ZC_TOOLCHAIN }} check --manifest-path ${{ matrix.manifest-path }} --target ${{ matrix.target }} --features "${{ matrix.features }}" --verbose --all-targets
```
Comment (joshlf) [inline .github/workflows/ci.yml:125]: Uber nit: Can you swap the flag order so it's `--all-targets --verbose` at the end?

**Review by joshlf — CHANGES_REQUESTED** [review]
Thanks! One uber nit, but otherwise looks good.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (joshlf) [reply .github/workflows/ci.yml:125]: Actually, another thought: Can you also add `--all-targets` to the `cargo build` step?

Comment (jswrenn) [reply .github/workflows/ci.yml]: Might I suggest scrapping the `cargo check` step entirely in favor of `cargo build --all-targets`?

Comment (joshlf) [reply .github/workflows/ci.yml]: Good question; I'll look into that and we can follow up in a separate PR if it turns out that's the way to go.

**Comment by joshlf** [comment]
Looks like the current errors are caused by not passing the right linker as described in https://github.com/rust-lang/rust/issues/28924. It may be worth just punting on this for `cargo build` and only doing it for `cargo check` unless there's an easy way to provide the linker flag that doesn't significantly increase the complexity of our CI tests or their configs.

**Comment by joshlf** [comment]
This is captured by our build matrix.

**Pull request closed without merging**